interface/orders/single_order_results.inc.php

```diff
@@ -332,358 +352,407 @@ function generate_order_report($orderid, $input_form = false, $genstyles = true,
+        let w = window.opener ? window.opener : window;
+        if (!top.tab_mode) {
+            for (; w.name != 'RTop' && w.name != 'RBot'; w = w.parent) {
+                if (w.parent == w) {
+                    // This message is not translated because a developer will need to find it.
+                    alert('Internal error locating target frame in ' + (window.opener ? 'opener' : 'window'));
+                    return false;
+                }
+            }
+            var othername = (w.name == 'RTop') ? 'RBot' : 'RTop';
+            w.parent.left_nav.forceDual();
+            w.parent.left_nav.loadFrame('pno1', othername, 'patient_file/summary/pnotes_full.php?orderid=' + encodeURIComponent(orderid));
+        } else {
+            let url = top.webroot_url + '/interface/patient_file/summary/pnotes_full.php?orderid=' + encodeURIComponent(orderid);
+            dlgopen(url, 'notes', 950, 750, false, '');
+        }
+        return false;
+    }
 
-var mypcc = '<?php echo $GLOBALS['phone_country_code'] ?>';
-
-// Called to show patient notes related to this order in the "other" frame.
-// This works even if we are in a separate window.
-function showpnotes(orderid) {
- // Find the top or bottom frame that contains or opened this page; return if none.
- var w = window.opener ? window.opener : window;
- if (!top.tab_mode) {
-     for (; w.name != 'RTop' && w.name != 'RBot'; w = w.parent) {
-         if (w.parent == w) {
-             // This message is not translated because a developer will need to find it.
-             alert('Internal error locating target frame in ' + (window.opener ? 'opener' : 'window'));
-             return false;
-         }
-     }
-     var othername = (w.name == 'RTop') ? 'RBot' : 'RTop';
-     w.parent.left_nav.forceDual();
-     w.parent.left_nav.loadFrame('pno1', othername, 'patient_file/summary/pnotes_full.php?orderid=' + encodeURIComponent(orderid));
- } else {
-     let url = '../../patient_file/summary/pnotes_full.php?orderid=' + orderid;
-     dlgopen(url, 'notes', 750, 500, '', '', {onClosed: 'reload'});
- }
- return false;
-}
-
-// Process click on LOINC code for patient education popup.
-function educlick(codetype, codevalue) {
-  dlgopen('<?php echo $GLOBALS['webroot']; ?>/interface/patient_file/education.php' +
-    '?type=' + encodeURIComponent(codetype) +
-    '&code=' + encodeURIComponent(codevalue) +
-    '&language=' + <?php echo js_url($language); ?>,
-    '_blank', 1024, 750,true); // Force a new window instead of iframe to address cross site scripting potential
-}
-
+    // Process click on LOINC code for patient education popup.
+    function educlick(codetype, codevalue) {
+        dlgopen(top.webroot_url + '/interface/patient_file/education.php' +
+            '?type=' + encodeURIComponent(codetype) +
+            '&code=' + encodeURIComponent(codevalue) +
+            '&language=<?php echo js_url($language); ?>',
```

Comment:
Was this line broken? (it should not of been broken)

Reply:
This was what was in current codebase.  I originally had it as urlencode. I just put to what was there and as you know, I don't seek out things to escape:)

Reply:
This is still different (not obvious, though) from the original (it will break if $language is not a number since the js escaping will place quotes around it). Here is the original line:
```
'&language=' + <?php echo js_url($language); ?>,
```